### PR TITLE
[sglang] fix: remove unused padding in SGLang rollout

### PR DESCRIPTION
**Pull request by PopSoda2002**
### What does this PR do?

What does this PR do?
There are some unused padding talked in this issue: 
https://github.com/zhaochenyang20/Awesome-ML-SYS-Tutorial/issues/193
- There are just 5 key fields which need to return back after rollout(example in `agent_loop`):
```python
batch = TensorDict(
{
    "prompts": prompt_ids,  # [bsz, prompt_length]
    "responses": response_ids,  # [bsz, response_length]
    "response_mask": response_mask,  # [bsz, response_length]
    "input_ids": input_ids,  # [bsz, prompt_length + response_length]
    "attention_mask": attention_mask,  # [bsz, prompt_length + response_length]
    "position_ids": position_ids, 
    # position_ids: [bsz, 3, prompt_length + response_length] or [bsz, prompt_length + response_length]
},
batch_size=len(inputs),
)
``` 
- Remove some unused variable like `prompt_loss_mask`
- Make `response_position_id` all zero tensor
- Copy class to avoid constructing a new class

### Test

`over_sample = 0.1` 
[wandb](https://wandb.ai/popsoda-university-of-washington/multi-turn-grpo-qwen2.5-3b-sglang/runs/1p87zi7v?nw=nwuserpopsoda)

image

No issue.

`over_sample = 0.0` 
[wandb](https://wandb.ai/popsoda-university-of-washington/multi-turn-grpo-qwen2.5-3b-sglang/runs/xloii5wm?nw=nwuserpopsoda)

image

As expected too

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`

### CODE REVIEW

**Review by zhaochenyang20 — APPROVED** [review]
Good to go. pass the CI plz

**Review by wuxibin89 — APPROVED** [review]
